### PR TITLE
Document Modules

### PR DESCRIPTION
**Pull request by mvz**
Fixes all the Style/Documentation offenses.

### CODE REVIEW

**Comment by mvz** [comment]
Ready for review. I'll squash afterwards.

File: lib/reek/smells/smell_detector.rb [hunk 1]
```diff
@@ -4,12 +4,6 @@
 
 module Reek
   module Smells
-    module ExcludeInitialize
-      def self.default_config
-        super.merge(EXCLUDE_KEY => ['initialize'])
-      end
-    end
-
```
Comment (troessner) [inline lib/reek/smells/smell_detector.rb]: YES.

File: lib/reek/spec.rb [hunk 2]
```diff
@@ -40,6 +40,72 @@ module Reek
...
+      ShouldReekOnlyOf.new(smell_category)
+    end
+
+    #
+    # Returns +true+ if and only if the target source code contains smells.
+    #
+    def reek
+      ShouldReek.new
+    end
   end
```
Comment (troessner) [inline lib/reek/spec.rb]: Good idea moving all of this here, way more concise!

**Comment by troessner** [comment]
Looks great, please squash!

**Comment by mvz** [comment]
Done!

**Pull request merged**